bzip2-sys/build.rs

```diff
@@ -25,6 +25,12 @@ fn main() {
 
     let dst = PathBuf::from(env::var_os("OUT_DIR").unwrap());
 
+    cfg.flag_if_supported("-ffunction-sections")
+        .flag_if_supported("-fdata-sections")
+        .flag_if_supported("-fmerge-all-constants")
+        .flag_if_supported("-Wl,--gc-sections")
+        .flag_if_supported("-Wl,--icf=safe");
```

Comment:
Makes sense to have `-ffunction-sections` and co. `-Wl,--gc-sections` and `-Wl,--icf=safe` can be omitted though as those only apply to linking by the cc crate, but we only build a staticlib here. They don't have any effect on linking done by rustc.